### PR TITLE
fix: respect offset and length on DbBytes#wrap

### LINKED ISSUE #16666: `DbBytes#getBytes` is incorrect when wrapped with offset
This test fails

```java
class DbBytesTest {
  @Test
  void shouldWrapBytes() {
    // given
    final var dbBytes = new DbBytes();
    final var bytes = new byte[] {1, 2, 3, 4, 5, 6, 7};
    final var buffer = new UnsafeBuffer(bytes);

    // when -- wrapping over part of the buffer
    dbBytes.wrap(buffer, 2, 3);

    // then
    assertArrayEquals(new byte[] {3, 4, 5}, dbBytes.getBytes());
  }
}
```

The cause is that `DbBytes#getBytes` simply returns the underlying byte array. This is incorrect when wrapping the `DbBytes` over a larger buffer with an offset. To support this, the returned value needs to be sliced for the correct offset and length.

### PR DESCRIPTION
**Pull request by megglos**
## Description

Aligned wrap implementation to DbString#wrap, it will create a copy on wrap which I considered the better compromise vs. doing it every time on get.

## Related issues

closes #16666 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by lenaschoenburg — APPROVED** [review]
Makes sense, thanks :+1:

File: zeebe/zb-db/src/test/java/io/camunda/zeebe/db/impl/DbByteTest.java [hunk 1]
```diff
@@ -47,4 +48,18 @@ public void shouldWrap() {
...
+    final var dbBytes = new DbBytes();
+    final var bytes = new byte[] {1, 2, 3, 4, 5, 6, 7};
+    final var buffer = new UnsafeBuffer(bytes);
+
+    // when -- wrapping over part of the buffer
+    dbBytes.wrap(buffer, 2, 3);
+
+    // then
+    assertThat(dbBytes.getBytes()).isEqualTo(new byte[] {3, 4, 5});
+  }
```
[the remark this answers is not in the document]
Comment (megglos) [reply zeebe/zb-db/src/test/java/io/camunda/zeebe/db/impl/DbByteTest.java:64]: lol overlooked the missing `s`

[the remark this answers is not in the document]
Comment (megglos) [reply zeebe/zb-db/src/test/java/io/camunda/zeebe/db/impl/DbByteTest.java:64]: Added a new test class also covering the other wrap method.

**Comment by korthout** [comment]
Thanks @megglos ❤️

**Pull request merged**